src/gen_smtp_client.erl

```diff
@@ -121,7 +123,46 @@ send_it_nonblock(Email, Options, Callback) ->
 			{ok, Receipt}
 	end.
 
--spec send_it(Email :: {string() | binary(), [string() | binary(), ...], string() | binary() | function()}, Options :: list()) -> binary() | {'error', any(), any()}.
+-opaque smtp_client_socket() :: {socket:socket(), list(), list()}.
```

Comment:
Maybe move this definition closer to the export?
Was also wondering if we shouldn't use a record here, for future changes?